lib/simulation/simulated_peripheral.dart

```diff
@@ -92,6 +96,7 @@ abstract class SimulatedPeripheral {
 
   Future<void> onConnect() async {
     _isConnected = true;
+    mtu = _negotiateMtu(mtu);
```

Comment:
We should add parsing of connect arguments and request the MTU passed by user. This can be extracted to separate issue.